authentication/src/main/kotlin/com/uber/sdk2/auth/internal/sso/UniversalSsoLink.kt

```diff
@@ -105,19 +105,15 @@ internal class UniversalSsoLink(
   private fun handleResult(result: ActivityResult): String {
     return when (result.resultCode) {
       RESULT_OK -> {
-        Log.d("xxxx", result.data?.getStringExtra("CODE_RECEIVED").orEmpty())
-        result.data?.let {
-          if (!it.getStringExtra("CODE_RECEIVED").isNullOrEmpty()) {
-            it.getStringExtra("CODE_RECEIVED").orEmpty()
-          } else if (!it.getStringExtra("EXTRA_ERROR").isNullOrEmpty()) {
-            throw AuthException.ClientError(it.getStringExtra("EXTRA_ERROR").orEmpty())
-          } else {
-            throw AuthException.ClientError(AuthException.EMPTY_RESPONSE)
-          }
+        result.data?.getStringExtra(EXTRA_CODE_RECEIVED)?.ifEmpty {
+          throw AuthException.ClientError(AuthException.AUTH_CODE_NOT_PRESENT)
         } ?: throw AuthException.ClientError(AuthException.NULL_RESPONSE)
       }
+
       RESULT_CANCELED -> {
-        throw AuthException.ClientError(AuthException.CANCELED)
+        result.data?.getStringExtra(EXTRA_ERROR)?.let {
```

Comment:
in which scenario this is populated?

also which EXTRA_ERROR is used here? is it from companion or android.provider.DocumentsContract.EXTRA_ERROR. If you intend to you your local version than maybe remove the one from android.provider

Reply:
It gets populated when an Uber app encounters an error in the SSO flow and returns it back to the client app.
It's used from companion. Will remove the unused import